File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = {}
        length = len(nums)
        for num in nums:
            count[num] = count.get(num, 0) + 1
            if count[num] > length / 2:
                return num
        

File: Data Structures & Algorithms/search-insert-position/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    searchInsert(nums: number[], target: number): number {
        let l = 0, r = nums.length
        while (l < r) {
            const m = l + Math.floor((r - l) / 2)
            if (target > nums[m]) {
                l = m + 1
            } else if (target < nums[m]) {
                r = m
            } else {
                return m
            }
        }
        return l
    }
}
